TestApp/Repositories/DataAccessRepositories.cs

```diff
@@ -0,0 +1,285 @@
+            try
+            {
+                if (specification.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.ToList();
+                    return persons;
+                }                
+                else
+                {
+                    var number = (int)((IdentificationType)Enum.Parse(typeof(IdentificationType), specification));
+                    var identities = ctx.Identifier.Distinct().Where(x => x.Value == number.ToString()).ToList();
+                    List<Person> personList = new List<Person>();
+                    if (identities != null)
+                    {
+                        foreach (var identity in identities)
+                        {
+                            Person person = new Person();
+                            person = ctx.Persons.Where(x => x.id == identity.Personid).FirstOrDefault();
+                            personList.Add(person);
+                        }
+                    }
+                                                            
+                    return personList;
+                }
+
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+        }
+
+        public bool CreatePersonWithIdetifiers(Person person)
+        {
+            try
+            {
+                ctx.Persons.Add(person);
+                ctx.SaveChanges();
+                return true;
+                //    int res = ctx.SaveChanges();
+                //    return res;
+            }
+            catch (Exception ex)
+            {
+                return false;
+            }
+        }
+        public bool CreatePersonWithOutIdetifiers(Person person)
+        {
+            try
+            {
+                Person personObj = new Person();
+                if (person != null)
+                {
+                    personObj.FirstName = person.FirstName;
+                    personObj.LastName = person.LastName;
+                    personObj.isDeleted = false;
+                    personObj.Identities = null;
+                    ctx.Persons.Add(personObj);
+                    ctx.SaveChanges();
+                    return true;
+                } else
+                {
+                    return false;
+                }
+                
+                
+                //    int res = ctx.SaveChanges();
+                //    return res;
+            }
+            catch (Exception ex)
+            {
+                return false;
+            }
+        }
+        public bool AddIdentifierToPerson(Guid id, int Value)
+        {
+            try
+            {
+                Person person = new Person();
+                person = ctx.Persons.Where(x => x.id == id && x.isDeleted==false).FirstOrDefault();
+                if (person != null)
+                {
+                    Identifier identifier = new Identifier();
+                    identifier.Value = Value.ToString();
```

Comment:
what about identifier type? its missing

Reply:
Its not missing there is no field for type in db